jstests/aggregation/bugs/server7695_isodates.js

```diff
@@ -0,0 +1,268 @@
+// SERVER-7695: Add $isoWeek, $isoWeekYear, and $isoDayOfWeek aggregation expressions.
+
+(function() {
+    "use strict";
+    var coll = db.server7695;
+    var testOpCount = 0;
+
+    coll.drop();
+
+    // Seed collection so that the pipeline will execute.
+    assert.writeOK(coll.insert({}));
+
+    // Helper for testing that op returns expResult.
+    function testOp(op, value, expResult) {
+        testOpCount++;
+        var pipeline = [{$project: {_id: 0, result: {}}}];
+        pipeline[0].$project.result[op] = value;
+        var msg = "Exptected {"+op+": "+value+"} to equal: "+expResult;
+        assert.eq(
+            coll.aggregate(pipeline).toArray()[0].result,
+            [{result: expResult}][0].result,
+            msg);
+    }
+
+    // this works
+    testOp('$dateToString', { date: new Date("1900-12-31T23:59:59Z"), format: "%V-%G"}, "01-1901")
+    // this is failing, but it shouldn't as it is the same as above, only rotated
+    testOp('$dateToString', { date: new Date("1900-12-31T23:59:59Z"), format: "%G-%V"}, "1901-01")
```

Comment:
@cswanson310 while creating some more tests for some more edge cases, I found a bug in the calculation of the isoWeek. Sadly I don't have any idea right now, why this is happening, because it only appears if I ask for the year before I ask for the week. The other way around it works perfectly fine.